src/util/log.d

```diff
@@ -92,32 +92,20 @@ struct Log
 {
     private Logger[] loggers;
 
-    // preallocate so we can fill it in @nogc fileDescriptors()
-    private int[] buffer;
-
     private uint levels;
 
     this(Logger[] loggers ...)
     in (loggers.all!"a !is null")
     {
         this.loggers = loggers.dup;
-        buffer = new int[this.loggers.length];
         levels = reduce!((a, b) => a | b.levels)(0, this.loggers);
     }
 
-    public int[] fileDescriptors() @nogc nothrow @safe
-    in (buffer.length >= loggers.length)
+    // Backdoor access for maintenance operations such as segfault handling.
+    // See `FileLogger.__fileDescriptor`.
+    public Logger[] __loggers() @nogc nothrow pure @safe
```

Comment:
This function just returns the loggers provided to the constructor:
how about saving them at the caller?

Reply:
Also works, but runs the risk of desyncing if for some reason `log` is changed again.